### PR TITLE
Fix AugmentedMemoizationPolicy max history bug. (3.0.x)

### PR DESCRIPTION
**Pull request by kedz**
**Proposed changes**:

Fixes three bugs with `AugmentedMemoizationPolicy`:

1. When more than max_history actions occurred in a row, information about the last user uttered event was incorrectly being discarded.
2. When a training story started with a user utterance AugmentedMemoizationPolicy would not be able to match it at test time if there were previous events before the initial user utterance event.
3. In the tests of `AugmentedMemoizationPolicy` in 3.0.x, the policy was not being constructed with the correct `max_history`, though its featurizer was, causing the policy not to truncate the tracker according to `max_history` actions in the unit tests.

For more discussion of the first two points see [this PR](https://github.com/RasaHQ/rasa/pull/10605) for the 2.8.x branch.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] ~updated the documentation~
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by kedz** [comment]
/modeltest

```yml
include:
 - dataset: ["all-core"]
   config: ["all-core"]

```

**Comment by kedz** [comment]
/modeltest

```yaml
include:
  - dataset: ["all-core"]
    config: ["all-core"]
```

File: tests/core/test_policies.py [hunk 1]
```diff
@@ -663,6 +664,89 @@ def test_augmented_prediction(
             model_storage=model_storage,
             resource=resource,
             execution_context=execution_context,
+            config={"max_history": max_history},
```
Comment (samsucik) [inline tests/core/test_policies.py:667]: Here and elsewhere we could use the `POLICY_MAX_HISTORY` constant instead of the `max_history` string.

**Review by samsucik — COMMENTED** [review]
I left only one minor comment but my comments from the 2.8 review apply here as well.

**Review by JEM-Mosig — APPROVED** [review]
Looks good to me :)

**Pull request merged**